### ISSUE TITLE
FNX-473 ⁃ Accessibility updates for Bottom Toolbar

### ISSUE THREAD

**Issue by mheubusch** [mozilla-mobile/fenix#1718]
This is specific to issues with Switch Access (accessibility function in Android) - The bottom toolbar means that switch users have to scan through the entire web page before they can access common functions

Please see Visual Design, Motion Design, and Interaction Design issues outlined here: https://docs.google.com/spreadsheets/d/1HvPFPQFtVVN0SDV97Qe6PBrDVhGQgxl8wBLpsXXvCm0/edit#gid=0

**Comment by lime124** [comment]
after talking with Eng this is the current state of this work:
All the other things are fixed except:
- [ ] the reader mode button (waiting for reader mode to be done)
- [ ] "incorrectly hiding when swiped" - need more info @shorlander 
- [ ] issues with Switch Access (accessibility function in Android) - The bottom toolbar means that switch users have to scan through the entire web page before they can access common functions

**Comment by vesta0** [comment]
UX and Eng will discuss how to resolve the accessibility issue.

**Comment by eeejay** [comment]
This is an issue with all toolbar functions, not just quick action items.

A user will have to scan through a variable number of items (depending on the web content), before reaching any browser function.

**Comment by ekager** [comment]
We have removed the QAB but should address the bottom toolbar issues

**Comment by sblatz** [comment]
It seems like all of the items on the browser toolbar are read through TalkBack. Users also now have the choice to have a top navigation bar (perhaps we should make this the default for users with accessibility settings?) The toolbar is also fixed when a screen reader is enabled so that users always have access to it.

@eeejay Can you help me understand what action can be taken on this?

**Comment by sblatz** [comment]
This also seems like a related issue to the bottom bar being broken for screen reader users: https://github.com/mozilla-mobile/fenix/issues/3467

**Comment by sblatz** [comment]
After a discussion with @MReschenberg (who looped in @asadotzler) we decided to do: 

1) Default to top toolbar if screen reader enabled on install
2) Warn users when toggling to bottom if they have screen reader enabled

@asadotzler can you verify that 2 is necessary / how to proceed with warning users that the bottom toolbar may cause issues with screen readers?

**Comment by sblatz** [comment]
Bumping @eeejay on this now that he's back 😄

**Comment by eeejay** [comment]
Hey! The top toolbar is great and really solves a big accessibility pain for many potential users.

Not sure what the outstanding question is. How to warn users? I don't know how we can do that well, I wouldn't bother adding more verbiage and confusion. Like you said, I think the default for switch (and, I think, TalkBack) users should be a top toolbar. 

I'm not sure setting it to a top default on install is good enough. What if the user got assistance in installing Firefox while the service wasn't running? Not sure how, but ideally if TalkBack/Switch is active, the toolbar should be on top, **unless** the user explicitly chooses the bottom one in preferences.

**Comment by sblatz** [comment]
Thanks for your feedback @eeejay. Based on my investigation I think I can do this:

1) If a user has accessibility services enabled and launches the app, we will set the default toolbar to be top (and FIXED not dynamic)

2) If a user enables accessibility services, fixed top toolbar will be enabled on the next app launch (unfortunately due to how we inflate the browser we only check if they're enabled when it's first created--do you think this is a major problem? It seems like this is a small edge case)

3) If a user with accessibility services goes into settings to change their toolbar it will respect what they select

**Comment by eeejay** [comment]
>     2. If a user enables accessibility services, fixed top toolbar will be enabled on the next app launch (unfortunately due to how we inflate the browser we only check if they're enabled when it's first created--do you think this is a major problem? It seems like this is a small edge case)

Yeah. I think that's good enough. I mentioned in the review: detecting the switch service will need some testing, it might not be as trivial as checking for explore by touch.

**Comment by sv-sdeiac** [comment]
Verified as fixed using the latest version of Firefox Preview Nightly 1/16 build #20160806.

Devices:
- Google Pixel (Android Q);
- Samsung Galaxy Note 8 (Android 9);
- Motorola Moto G 6 (Android 8.0).
